src/plot_api/plot_api.js

```diff
@@ -2586,7 +2586,7 @@ function makePlotFramework(gd) {
         .attr('id', 'defs-' + fullLayout._uid);
 
     fullLayout._topdefs = fullLayout._toppaper.append('defs')
-        .attr('id', 'defs-' + fullLayout._uid);
+        .attr('id', 'topdefs-' + fullLayout._uid);
```

Comment:
svg ids are treated as regular DOM ids. They are not scoped to one particular `<svg>`: 

http://codepen.io/etpinard/pen/ONWVaL